tests/TestCase/File/Transformer/Base64TransformerTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Josegonzalez\Upload\Test\TestCase\File\Transformer;
+
+use Cake\ORM\Entity;
+use Cake\ORM\Table;
+use Cake\TestSuite\TestCase;
+use Josegonzalez\Upload\File\Transformer\Base64Transformer;
+use VirtualFileSystem\FileSystem as Vfs;
+
+class Base64TransformerTest extends TestCase
+{
+    public function setup()
+    {
+        $this->entity = $this->getMockBuilder('Cake\ORM\Entity')->getMock();
+        $this->table = $this->getMockBuilder('Cake\ORM\Table')->getMock();
+        $this->data = ['data' => 'Y2FrZXBocA==', 'name' => '5a2e69ff-c2c0-44c1-94a7-d791202f0067.txt'];
```

Comment:
We should keep in mind that using a custom array format like this makes validation tricker. Most validation methods for uploaded files won't be usable. I haven't reviewed the full PR but you need to ensure proper validation is done for this array format and someone can't sneak in whatever they want using this feature.